File: src/main/java/com/yerti/core/enchantmenets/EnchantmentHandler.java
package com.yerti.core.enchantmenets;

import org.bukkit.entity.Player;
import org.bukkit.event.EventHandler;
import org.bukkit.event.Listener;
import org.bukkit.event.entity.EntityDamageByEntityEvent;
import org.bukkit.inventory.ItemStack;

import java.util.ArrayList;
import java.util.List;

public class EnchantmentHandler implements Listener {

    public static List<CombatEnchant> enchants = new ArrayList<>();

    @EventHandler
    public void onHit(EntityDamageByEntityEvent event) {

        if (event.getDamager() instanceof Player) {
            Player player = (Player) event.getDamager();

            ItemStack stack = player.getItemInHand();

            for (CombatEnchant enchant : enchants) {
                if (stack.containsEnchantment(enchant)) {
                    enchant.sendEvent(event);
                }
            }

        }

    }


}

File: src/main/java/com/yerti/core/enchantmenets/EnchantmentHandler.kt
package com.yerti.core.enchantmenets

import org.bukkit.entity.Player
import org.bukkit.event.EventHandler
import org.bukkit.event.Listener
import org.bukkit.event.entity.EntityDamageByEntityEvent
import java.util.*

class EnchantmentHandler : Listener {
    @EventHandler
    fun onHit(event: EntityDamageByEntityEvent) {
        if (event.damager is Player) {
            val player = event.damager as Player
            val stack = player.itemInHand
            for (enchant in enchants) {
                if (stack.containsEnchantment(enchant)) {
                    enchant.sendEvent(event)
                }
            }
        }
    }

    companion object {
        var enchants: List<CombatEnchant> = ArrayList()
    }
}